### PR TITLE
feat(deploy-upgrade): add deploy and upgrade scripts for `GroupingWorkflows`

### LINKED ISSUE #49: Update Deploy and Upgrade Scripts to include `GroupingWorkflows`
## Description and Context
There currently isn't a script to deploy or upgrade the `GroupingWorkflows` contract.

## Suggested Solution
- Update the deployment and upgrade scripts to include `GroupingWorkflows`.
- Test to ensure proper integration.

## Definition of Done
- `GroupingWorkflows` is successfully integrated, tested, and functional within the deployment and upgrade scripts.

### PR DESCRIPTION
**Pull request by sebsadface**
## Description
This PR introduces deployment and upgrade logic for the `GroupingWorkflows` contract.

**Key changes include**:
- Added deployment logic for `GroupingWorkflows` contract to the deployment script (`Main.s.sol`).
- Added a separate deployment script for the `MockRewardPool` contract.
- Created an upgrade script for `GroupingWorkflows`.
- Updated the SPGNFT deploy and upgrade scripts to include a parameter for the `GroupingWorkflows` address.
- Reorganized the `scripts` folder by adding `deployment` and `upgrade` directories, moving relevant scripts accordingly.

## Test Plan
All existing and new deployment and upgrade scripts have been tested on the testnet, confirming that all scripts operate as expected. For details on the test results see the dropdown section below:

<details>
  <summary>Test Results</summary>

### Deployment Testing

  1. **Deployment Script (`Main.s.sol`)**:  
     The deployment script was run with seed `9032024` on the partner testnet, successfully deploying and verifying contracts at the following addresses:

     ```json
     {
       "main": {
         "GroupingWorkflows": "0x3E89688EEae8CaC9bd5a27367351D6614E3E721F",
         "SPG": "0x11a3b6181cbfEDF77ce249953eF76BBD2108EB98",
         "SPGNFTBeacon": "0xE1a6AE889Df8842Faf332F93cb1c47B4820452FE",
         "SPGNFTImpl": "0xe7D443CDfDBa013085D0Abe6ECbB0cC79fA72221"
       }
     }
     ```

     These contracts can be verified on the block explorer:
     - [SPG](https://explorer.testnet.storyprotocol.net/address/0x11a3b6181cbfedf77ce249953ef76bbd2108eb98)
     - [GroupingWorkflows](https://explorer.testnet.storyprotocol.net/address/0x3E89688EEae8CaC9bd5a27367351D6614E3E721F?tab=contract)
     - [SPGNFTBeacon](https://explorer.testnet.storyprotocol.net/address/0xE1a6AE889Df8842Faf332F93cb1c47B4820452FE)
     - [SPGNFT](https://explorer.testnet.storyprotocol.net/address/0xe7D443CDfDBa013085D0Abe6ECbB0cC79fA72221)

  2. **Contract Config**:
     - Successfully called `setNftContractBeacon` for both GroupingWorkflows and SPG. Transaction details:
       - [GroupingWorkflows](https://explorer.testnet.storyprotocol.net/tx/0x1f080ffe748db231172e9d905d7d31a3ed63ceff19bcd2a73feb5032589177b2)
       - [SPG](https://explorer.testnet.storyprotocol.net/tx/0x690b3be0965ba408ebb8c8eda501a27b436c9be9786ec253f5777a468e48689d)

### Upgrade Testing

  1. **SPG Upgrade**:  
     A new SPG implementation was deployed and verified on the testnet at `0x211C760754C1F4969F95fB1E2e3A7c927EfCA041`. See the contract details [here](https://explorer.testnet.storyprotocol.net/address/0x211C760754C1F4969F95fB1E2e3A7c927EfCA041).  
     - Successfully called `upgradeToAndCall` on the SPG proxy contract to point to the new implementation. [Transaction details](https://explorer.testnet.storyprotocol.net/tx/0xf7ecce1622063e564e2eb4aef2aa86b5dc716aca6407c704279e3eff9bfb2aae).

  2. **GroupingWorkflows Upgrade**:  
     A new GroupingWorkflows implementation was deployed and verified on the testnet at `0xBD598Db8D688Ab4488eC45dcb6238D7fB28700e6`. See the contract details [here](https://explorer.testnet.storyprotocol.net/address/0xBD598Db8D688Ab4488eC45dcb6238D7fB28700e6).  
     - Successfully called `upgradeToAndCall` on the GroupingWorkflows proxy contract to point to the new implementation. [Transaction details](https://explorer.testnet.storyprotocol.net/tx/0x815a63d17acd63b6720ee0b7577026722b1875a0b6ec5bd95f6d7b9704ff8eb3).

  3. **SPGNFT Upgrade**:  
     A new SPGNFT implementation was deployed and verified on the testnet at `0xBD961c2894B3E39af256CE6C6A99F246c6C0991C`. See the contract details [here](https://explorer.testnet.storyprotocol.net/address/0xBD961c2894B3E39af256CE6C6A99F246c6C0991C).  
     - Successfully called `upgradeCollections` on the SPG proxy contract to point SPGNFT to the new implementation. [Transaction details](https://explorer.testnet.storyprotocol.net/tx/0x65e61410832c93e4f3e754bd82824904e7a099bf049fc439ef854ea5dd546a8e).

</details>

## Related Issue

This PR closes #49

### CODE REVIEW

File: script/deployment/MockRewardPool.s.sol [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// test
+import { MockEvenSplitGroupPool } from "test/mocks/MockEvenSplitGroupPool.sol";
+
+contract MockRewardPool is Script, BroadcastManager, JsonDeploymentHandler {
+    using stdJson for string;
+
+    constructor() JsonDeploymentHandler("main") {}
+
+    /// @dev To use, run the following command (e.g. for Sepolia):
+    /// forge script script/deployment/MockRewardPool.s.sol:MockRewardPool --rpc-url $RPC_URL --broadcast --verify -vvvv
```
Comment (kingster-will) [inline script/deployment/MockRewardPool.s.sol:21]: It would better give example of deploying on public testnet.

**Review by kingster-will — COMMENTED** [review]
LGTM. 
It would be better mention the test procedures and result of upgrade scripts.

File: script/deployment/MockRewardPool.s.sol (hunk 1 above)
Comment (sebsadface) [reply script/deployment/MockRewardPool.s.sol]: Resolved in [`f95ddd8`](https://github.com/storyprotocol/protocol-periphery-v1/pull/50/commits/f95ddd8a89f9901ce2f634e9f4388e4d986b8b92) :)

**Comment by sebsadface** [comment]
> LGTM. It would be better mention the test procedures and result of upgrade scripts.

^Updated PR description to include test results :)

**Review by kingster-will — APPROVED** [review]

**Pull request merged**